### ISSUE TITLE
Isn't `ConstStringListRef` highly inflexible or am I missing something?

### ISSUE THREAD

**Issue by blackpill0w** [ArthurSonzogni/FTXUI#843]
Hi,

I am trying create a `Menu` but the items I have are not `std::string`, I have, let's say, a list of `struct A` and I want a field of the struct (which is a string) in the menu, do I need to keep a copy of each one to create a vector of std::string for the Menu?
I can't even use `std::vector<char *>` which would be way better.
Maybe we could add something like this
```c++
Component Menu(std::vector<T> entries, /* other stuff */, std::function<std::string(T&)> convertEntriesToString);
```
to convert entries to string each time you render,

thanks.

**Comment by ArthurSonzogni** [comment]
This might be a bit costly to call the functor everytime we need to observe the list size.

I was thinking adding an option to ConstStringListRef to provide class implementing the required functions.

```cpp
class ConstStringListRefAdapter {
  virtual int Size() = 0;
  virtual const std::string Entry(int index) = 0;
}

ConstStringListRef(ConstStringListRefAdapter* adapter);
ConstStringListRef(std::unique_ptr<ConstStringListRefAdapter> adapter);
```

But I am not totally convinced.

**Comment by blackpill0w** [comment]
You're right the `ConstStringListRefAdapter` stills creates a new string each time so that would very bad for performance, what about a `vector<std::string_view>`? You still have the hit when you compute the size but it's better I think.
I also thought of using `vector<std::string*>`, I don't know how much change would be required to the existing code, so tell me what you think about the two suggestions.
